atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/InitialisingSweeperService.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ForwardingObject;
+
+public final class InitialisingSweeperService extends ForwardingObject implements SweeperService {
```

Comment:
If all of these classes just forward all methods to the delegate (and don't have any other special logic), we could probably create a utility class using `Proxy` and `AbstractInvocationHandler` to handle the logic of checking if the delegate is initialized, and forwarding all calls to it